service/controller/resource/clusterrolebinding/update.go

```diff
@@ -21,7 +23,24 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange inte
 		// Create the cluster role bindings in the Kubernetes API.
 		for _, clusterRoleBinding := range clusterRoleBindingsToUpdate {
 			_, err := r.k8sClient.RbacV1().ClusterRoleBindings().Update(clusterRoleBinding)
-			if err != nil {
+			if isExternalFieldImmutableError(err) {
+				// We can't change a RoleRef, so delete the old CRB and replace it
```

Comment:
Maybe we should add a `log` message here as well.

Reply:
Good idea - added some